prime-router/src/main/kotlin/Report.kt

```diff
@@ -72,7 +87,11 @@ class Report : Logging {
     /**
      * the UUID for the report
      */
-    val id: ReportId
+    // TODO: Made this var instead of val so we can update the report ID after creation in the async process
+    //  functionality. There is a way to do it as a passed in variable, but the way we create reports via parsing
+    //  contentBody does not lend itself to that way of doing it. Once we are on FHIR internal spec, this could be
+    //  changed back to val - CD
+    var id: ReportId
```

Comment:
This was a way to minimize changes to existing structure for the async endpoint - working on changing this to save the generated internal CSV and load that back instead of regenerating the Report from the original received file.

Reply:
Updated to use .copy instead, not sure if it is better - now it creates an extra copy of the in-memory report which we then use to copy to a new one and set the existing ID. This will all change anyway when we switch to internal FHIR format

Reply:
After discussion with team members I am going to leave it as it is, and not use .copy.  It is tech debt, added numerous comments to indicate this.
Overall it reduces our performance improvements to need to re-create the report in memory again just to copy it to get the id. The structure of the code precludes passing in the existing ID in the first creation since our serializers are both parsing AND creating the report.